### PR TITLE
Add retry to GetConnectedWebSocket

### PR DESCRIPTION
**Pull request by stephentoub**
To hopefully avoid spurious failures when connecting to the remote endpoint.

cc: @cipop, @davidsh, @ericeil

### CODE REVIEW

**Comment by stephentoub** [comment]
Test Innerloop OSX Release Build and Test please

**Comment by davidsh** [comment]
LGTM

File: src/System.Net.WebSockets.Client/tests/WebSocketHelper.cs [hunk 1]
```diff
@@ -67,27 +67,47 @@ public static class WebSocketHelper
             ITestOutputHelper output,
             TimeSpan keepAliveInterval = default(TimeSpan))
         {
-            var cws = new ClientWebSocket();
-            if (keepAliveInterval.TotalSeconds > 0)
-            {
-                cws.Options.KeepAliveInterval = keepAliveInterval;
-            }
+            const int MaxTries = 5;
+            int BetweenTryDelaySeconds = 1000;
```
Comment (stephentoub) [inline src/System.Net.WebSockets.Client/tests/WebSocketHelper.cs]: Just noticed this should be named Milliseconds. Will fix.

**Comment by bartonjs** [comment]
LGTM (with Seconds => Milliseconds)

File: src/System.Net.WebSockets.Client/tests/WebSocketHelper.cs (hunk 1 above)
Comment (jasonwilliams200OK) [inline src/System.Net.WebSockets.Client/tests/WebSocketHelper.cs]: nit: (this being local variable) should it be `betweenTryDelayMilliseconds` instead?

Comment (stephentoub) [inline src/System.Net.WebSockets.Client/tests/WebSocketHelper.cs]: Yup, fixed, thanks.  I initially had it as a const, hence the name, but then changed it when I added the backoff and neglected to rename as well.

**Comment by stephentoub** [comment]
@mmitche, OSX is failing on various PRs now due to https://github.com/dotnet/corefx/issues/9066: 0 errors, and then fails at the very end with:
```
[xUnit] [ERROR] - The plugin hasn't been performed correctly: Problem on deletion
```

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Debug Build and Test please
Test Innerloop OSX Release Build and Test please

**Pull request merged**